ui/pages/confirmations/confirmation/confirmation.js

```diff
@@ -310,6 +318,7 @@ export default function ConfirmationPage({
     trackEvent,
     isSnapDialog,
     snapName,
+    networkConfigurationsByChainId,
```

Comment:
This might cause performance issues and render the useMemo useless. To prevent that we might want to use `createDeepEqualSelector` util for the `getNetworkConfigurationsByChainId` selector.

Explanation:
Everytime a controller state (any controller) is updates its state, those background updates are reflected in the UI redux state, causing the `networkConfigurationsByChainId` to be updated as well (values might the same but the reference is different). React uses Object.is on the dependencies array comparison to decide if it needs to re-run the effect or memo. And in this case as the reference changed, so react will re-run the useMemo. To prevent it we need to do deepEqual equality check. The `createDeepEqualSelector` gives us that plus the reselect default memoization.

Reply:
memoized in https://github.com/MetaMask/metamask-extension/pull/26433/commits/b2ed30bb328743bc957e1243707b1c347043afbb